### DISCUSSION TITLE
No candidates in ruby files

### DISCUSSION THREAD

**Discussion by idr4n** [company-mode/company-mode#1177] [Q&A] [answered]
Hi,

I recently installed company but I am not getting any completion candidates in my ruby files. I do get however company-dabbrev completions. However, company gives me candidates flawlessly in irb through inf-ruby.

I have tried both ruby-mode and enh-ruby-mode, but in both cases the situation is the same (no completion candidates in ruby files). After installing robe I ddid get completion candidates once I start it with `robe-start`.

I am running mac and managing packages with cask-pallete. Thanks in advance!

**Comment by dgutov** [comment]
`M-x company-diag` says?

**Comment by idr4n** [comment]
I get the following:

```
Emacs 25.1.50.1 (x86_64-apple-darwin15.5.0) of 2016-07-16 on iMac.local
Company 0.9.0

company-backends: (company-bbdb company-nxml company-css company-eclim company-semantic company-clang company-xcode company-cmake company-capf company-files
  (company-dabbrev-code company-gtags company-etags company-keywords)
  company-oddmuse company-dabbrev)

Used backend: company-dabbrev
Prefix: nil
Completions: none
```

I forgot to mention that I do get completion candidates of the variables created in the ruby file, but not completion candidates of ruby methods, etc.

**Comment by dgutov** [comment]
You mean standard library methods? Or from gems and such, not from open files?

If so, why would you expect to get them?

**Comment by idr4n** [comment]
yes, standard library methods (e.g. include?, upcase, etc.). I do get them in inf-ruby but not in the current ruby file.

**Comment by dgutov** [comment, marked as answer]
You get them in `inf-ruby` because a Ruby process is available, and Ruby has good introspection capabilities. Without a Ruby process, or an advanced completion backend, `company-mode` doesn't know about Ruby methods.

Someone could try to create a backend that uses `inf-ruby` completion capabilities to offer completions in the source files, but nobody has done that so far. And Robe pretty much already does that (and several other things as well).

**Comment by idr4n** [comment]
Got it, I understand now why I don't get completion suggestions. Thanks for the time!
